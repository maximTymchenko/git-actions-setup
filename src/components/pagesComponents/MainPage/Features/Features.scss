@import '../../../../styles/vars';

.features {
  &__container {
    position: relative;
    bottom: 50px;
    left: 10px;
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    max-width: 510px;
    text-align: center;
    @include main__padding;

    @media screen and (max-width: $desktop) {
      max-width: 490px;
    }
  }
  &__wrapper {
    background-image: url('../../../../assets/img/dots-bg.svg');
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center center;
    display: flex;
    justify-content: center;

    @media screen and (max-width: $tablet) {
      background: none;
    }
  }
}
