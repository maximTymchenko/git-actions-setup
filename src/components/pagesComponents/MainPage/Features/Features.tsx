import React, { FC } from 'react';
import './Features.scss';
import arrowRight from '../../../../assets/img/arr-right.svg';
import SectionTitle from '../../../common/SectionTitle';
import Button from '../../../common/Button';
import { Link } from 'gatsby';

const Features: FC = () => {
    return (
        <div data-testid="features" className="features__wrapper">
            <div className="features__container">
                <SectionTitle
                    type="secondary"
                    className="title"
                    title="More amazing services and success stories awaiting you here"
                />

                <Link to="/what-we-do/" className="link">
                    <Button type="secondary" subtype="mt" desc="Discover all" icon={arrowRight}>
                        Discover all
                    </Button>
                </Link>
            </div>
        </div>
    );
};

export default Features;
