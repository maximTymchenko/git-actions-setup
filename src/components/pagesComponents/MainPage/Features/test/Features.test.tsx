import React from 'react';
import Features from '../Features';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('renders elem correctly', () => {
    const { getByTestId } = render(<Features></Features>);
    expect(getByTestId('features')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer.create(<Features></Features>).toJSON();
    expect(tree).toMatchSnapshot();
});
