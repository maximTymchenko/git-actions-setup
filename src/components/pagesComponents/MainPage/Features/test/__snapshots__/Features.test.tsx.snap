// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="features__wrapper"
  data-testid="features"
>
  <div
    className="features__container"
  >
    <h2
      className="title title--secondary"
      data-testid="section-title"
    >
      More amazing services and success stories awaiting you here
    </h2>
    <a
      className="link"
      href="/what-we-do/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <button
        className="button button--secondary button--undefined button--mt"
        data-testid="button"
      >
        Discover all
        <img
          alt="Discover all"
          className="button__icon button__icon--undefined"
          data-testid="button__icon"
          src="test-file-stub"
        />
      </button>
    </a>
  </div>
</div>
`;
